### ISSUE TITLE
Discrepancy in pulling specific ml-plugin dependency version while building ES 8.13.3 and ES 8.14.1

### ISSUE THREAD

**Issue by amolai** [linux-on-ibm-z/scripts#23]
Hi Team,

I am trying to build an image of Elastic version 8.13.3 and 8.14.1 for s390x, and below is my observation 

1. While building 8.13.3, if for ml-plugin I use version latest (8.15.1), the build is successful. However if I specify a particular version like 8.13.3 or lower. In that I get the following error.
`What went wrong:
Could not determine the dependencies of task ':x-pack:plugin:ml:explodedBundlePlugin'.
> Could not resolve all task dependencies for configuration ':x-pack:plugin:ml:nativeBundle'.
   > Could not find org.elasticsearch.ml:ml-cpp:8.13.3-SNAPSHOT.
     Searched in the following locations:
       - https://artifacts-snapshot.elastic.co/ml-cpp/8.13.3-SNAPSHOT/downloads/ml-cpp/ml-cpp-8.13.3-SNAPSHOT-deps.zip
     Required by:
         project :x-pack:plugin:ml
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org/.`

Randomly tried the same for lower versions like 7.17.24. But get similar dependency error. 
Surprisingly, similar version specifications work for version 8.14.1 as described below
2. While build 8.14.1, if the ml-plugin version is latest or 8.14.1 or 8.13.3, the build is successful. 

I am trying to build a dockerfile which could be easily changed to upgrade the version or in some cases downgrade as well. Hence, trying to understand this behaviour. 

To summarise: We can specify any ml-plugin version for 8.14.1, but same is not true for version 8.13.3, 8.13.3 works only with latest.

Note: Referring the build script to create a dockerfile in the directory of these repos 
1. 8.13.3: https://github.com/linux-on-ibm-z/scripts/tree/master/Elasticsearch/8.13.3
2. 8.14.1: https://github.com/linux-on-ibm-z/scripts/tree/master/Elasticsearch/8.14.1
